src/Microsoft.VisualStudio.ProjectSystem.Managed/Tree/Dependencies/Subscriptions/RuleHandlers/ProjectRuleHandler.cs

```diff
@@ -39,7 +39,8 @@ internal class ProjectRuleHandler : DependenciesRuleHandlerBase
         public ProjectRuleHandler(
             IAggregateDependenciesSnapshotProvider aggregateSnapshotProvider,
             DependenciesSnapshotProvider snapshotProvider,
-            IUnconfiguredProjectCommonServices commonServices)
+            IUnconfiguredProjectCommonServices commonServices,
+            [Import(AllowDefault = true)] ISolutionService? solutionService)
```

Comment:
Why is this optional?

Reply:
Because of the split between VS/non-VS layers. `ISolutionService` is defined in `MS.VS.ProjectSystem.Managed`. `SolutionService` is defined in `MS.VS.ProjectSystem.Managed.VS`.